Improve the documentation for two memory counters (#23126)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties.rst

* **Type:** ``integer``
* **Default value:** ``38``

  Specifies the total amount of memory in GB that can be used for all queries on a worker node.
  Memory for system usage such as disk spilling and cache prefetch are not
  counted in this query memory usage.

``query-reserved-memory-gb``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``4``

  Specifies the total amount of memory in GB reserved for the queries on
  a worker node. A query can only allocate from this reserved space if

>   1) the non-reserved space in "query-memory-gb" is used up; and 2) the amount

Reviewer: ```suggestion
  1) the non-reserved space in ``query-memory-gb`` is used up; and 2) the amount
```

>   it is trying to get is less than 'memory-pool-reserved-capacity'.

Reviewer: ```suggestion
  it is trying to get is less than ``memory-pool-reserved-capacity``.
```

``system-memory-gb``
^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``40``

  Memory allocation limit enforced via internal memory allocator. It consists of two parts:

>   1) Memory used by the queries as specified in "query-memory-gb"; 2) System memory usage,

Reviewer: ```suggestion
  1) Memory used by the queries as specified in ``query-memory-gb``; 2) System memory usage,
```
